nodes.tf

```diff
@@ -11,175 +11,319 @@ locals {
       additional_node_labels = ""
     },
     monitoring-workers = {
-      min_size         = 1
-      max_size         = 4
-      instance_types   = ["m5.large"]
-      type             = "cpu"
-      spot             = false
-      desired_capacity = "1"
-      taints           = "--register-with-taints=indico.io/monitoring=true:NoSchedule"
+      min_size               = 1
+      max_size               = 4
+      instance_types         = ["m5.large"]
+      type                   = "cpu"
+      spot                   = false
+      desired_capacity       = "1"
+      taints                 = "--register-with-taints=indico.io/monitoring=true:NoSchedule"
       additional_node_labels = ""
     },
     weaviate = {
-      type             = "cpu"
-      spot             = false
-      instance_types   = ["r5a.xlarge"]
-      min_size         = 1
-      max_size         = 3
-      desired_capacity = "3"
-      taints           = "--register-with-taints=indico.io/weaviate=true:NoSchedule"
+      type                   = "cpu"
+      spot                   = false
+      instance_types         = ["r5a.xlarge"]
+      min_size               = 1
+      max_size               = 3
+      desired_capacity       = "3"
+      taints                 = "--register-with-taints=indico.io/weaviate=true:NoSchedule"
       additional_node_labels = ""
     },
     weaviate-workers = {
-      type             = "cpu"
-      spot             = false
-      instance_types   = ["c6a.2xlarge"]
-      min_size         = 1
-      max_size         = 4
-      desired_capacity = "2"
-      taints           = "--register-with-taints=indico.io/weaviate-workers=true:NoSchedule"
+      type                   = "cpu"
+      spot                   = false
+      instance_types         = ["c6a.2xlarge"]
+      min_size               = 1
+      max_size               = 4
+      desired_capacity       = "2"
+      taints                 = "--register-with-taints=indico.io/weaviate-workers=true:NoSchedule"
       additional_node_labels = ""
     }
   }
 
   standalone_node_groups = {
     default-workers = {
-      min_size         = 1
-      max_size         = 3
-      instance_types   = ["m5.xlarge"]
+      min_size               = 1
+      max_size               = 3
+      instance_types         = ["m5.xlarge"]
+      type                   = "cpu"
+      spot                   = false
+      desired_capacity       = "0"
+      additional_node_labels = ""
+      taints                 = ""
+    }
+  }
+
+  karpenter_node_group = {
+    karpenter = {
       type             = "cpu"
       spot             = false
-      desired_capacity = "0"
-      additional_node_labels = ""
-      taints           = ""
+      instance_types   = ["t3.medium"]
+      min_size         = 1
+      max_size         = 1
+      desired_capacity = "1"
+      taints           = "--register-with-taints=node-role.kubernetes.io/control-plane:NoSchedule"
+    }
+  }
+
+  intake_default_node_pool = {
```

Comment:
Is there any other way to handle this? The near duplication of node_groups and node_pools feels like this is going to be very error prone when making changes.

Reply:
I will try to map the node_group to node_pool. The issue here is to handle the taints and labels, but I will try

Reply:
All set now, using only node_groups